api/src/main/scala/org/ensime/api/incoming.scala

```diff
@@ -19,8 +19,14 @@ sealed trait RpcRequest
 sealed trait RpcStartupRequest extends RpcRequest
 
 /**
+ * Respond with a `ConnectionInfo`.
+ */
+case object AsynConnectionInfoReq extends RpcStartupRequest
```

Comment:
oh, no I was thinking there would be no request for this. It would just be provided on connection.

We could have a system property or something (on by default) that does the current (legacy) behaviour so we don't break the clients.

Reply:
Ah, ok. I'll rewrite it then. :)